pkg/appgw/requestroutingrules.go

```diff
@@ -146,16 +154,18 @@ func (c *appGwConfigBuilder) noRulesIngress(cbCtx *ConfigBuilderContext, ingress
 				PathRules:                  &[]n.ApplicationGatewayPathRule{},
 			},
 		}
+		(*listenerIngress)[listenerID] = ingress
 	}
 }
 
-func (c *appGwConfigBuilder) getPathMaps(cbCtx *ConfigBuilderContext) map[listenerIdentifier]*n.ApplicationGatewayURLPathMap {
+func (c *appGwConfigBuilder) getPathMaps(cbCtx *ConfigBuilderContext) (map[listenerIdentifier]*n.ApplicationGatewayURLPathMap, map[listenerIdentifier]*networking.Ingress) {
 	urlPathMaps := make(map[listenerIdentifier]*n.ApplicationGatewayURLPathMap)
+	listenerIngress := make(map[listenerIdentifier]*networking.Ingress)
```

Comment:
there can be many ingresses that tie up to the same listener. In following code, you will always take the last ingress and pass it up. in case the `ingress` order changes, it is possible that a different rewrite rule set is passed up to the rule. This will lead to an unstable configuration which might keep filling between two rule sets

Reply:
can you check `getDefaultFromRule` which returns the rule level properties. It is currently calculating the backend pool and backend http settings. You can modify it to also calculate the rewrite rule set that will be applied at the rule level.